### PR TITLE
Implement AMP support for Invibes

### PR DESCRIPTION
**Pull request by muuki88**
I hope this is the correct approach to check if a request is acutally an AMP request.

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java [hunk 1]
```diff
@@ -121,6 +121,15 @@ private void validateImp(Imp imp) {
         }
     }
 
+    private boolean isAmp(BidRequest request) {
+        if (request.getExt() == null) {
+            return false;
+        } else if (request.getExt().getPrebid() == null) {
+            return false;
+        }
+        return request.getExt().getPrebid().getAmp() != null;
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java:130]: Not our way of doing such things, we prefer solutions like in `resolveConsentString` or wrapping with `Optional` and then chaining `.map` calls.

File: src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java [hunk 2]
```diff
@@ -278,6 +281,55 @@ public void makeHttpRequestsShouldCreateInvibesBidRequestWithCorrectParams() thr
...
+                .lid(BUYER_UID)
+                .bvid("test")
+                .width(String.valueOf(DEVICE_W))
+                .height(String.valueOf(DEVICE_H))
+                .build();
+
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(HttpRequest::getPayload)
+                .containsOnly(expectedRequest);
```
Comment (And1sS) [inline src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java:298]: There is no need to check for size if you use `.containsExactly` at the end

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java (hunk 1 above)
Comment (muuki88) [reply src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java]: Oh nice :tada: ! Last time I used `Optional` it was marked in the code review.

File: src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java (hunk 2 above)
Comment (muuki88) [reply src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java:298]: I copied the test from above and changed only the check for `isAMP`. Should I resolve this?

Comment (muuki88) [reply src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java]: done for both

File: src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java (hunk 1 above)
Comment (And1sS) [reply src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java]: Yeah, we decided to start using `Optional` recently.

File: src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java (hunk 2 above)
Comment (And1sS) [reply src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java]: Would be really nice.

**Review by And1sS — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java [hunk 3]
```diff
@@ -121,6 +124,13 @@ private void validateImp(Imp imp) {
         }
     }
 
+    private boolean isAmp(BidRequest request) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java:127]: Seems this one can be static

File: src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java [hunk 4]
```diff
@@ -291,7 +295,55 @@ public void makeHttpRequestsShouldCreateInvibesBidRequestWithCorrectParams() thr
...
+                .invibBVLog(true)
+                .videoAdDebug(true)
+                .lid(BUYER_UID)
+                .bvid("test")
+                .width(String.valueOf(DEVICE_W))
+                .height(String.valueOf(DEVICE_H))
+                .build();
+
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java:344]: Should we check the whole request or only `isAmp` flag?

Comment (muuki88) [reply src/test/java/org/prebid/server/bidder/invibes/InvibesBidderTest.java:344]: I opted for checking the entire request as it should stay the same with only the AMP flag changed

File: src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java (hunk 3 above)
Comment (muuki88) [reply src/main/java/org/prebid/server/bidder/invibes/InvibesBidder.java]: done :)

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**